src/common/core/coretree.js

```diff
@@ -695,29 +694,19 @@ define([
         };
 
         this.getKeys = function (node, predicate) {
-            ASSERT(typeof predicate === 'undefined' || typeof predicate === 'function');
-
+            var result;
             node = self.normalize(node);
-            predicate = predicate || __noUnderscore;
 
             if (typeof node.data !== 'object' || node.data === null) {
                 return null;
             }
 
-            var keys = Object.keys(node.data),
-                result = [],
-                i = keys.length;
-
-            while (--i >= 0) {
-                if (predicate(keys[i])) {
-                    result.push(keys[i]);
-                }
-            }
+            result = this.getRawKeys(node.data, predicate);
             return result;
         };
 
         this.getRawKeys = function (object, predicate) {
-            ASSERT(typeof predicate === 'undefined' || typeof predicate === 'function');
+            ASSERT(predicate === undefined || typeof predicate === 'function');
```

Comment:
is this ASSERT really necessary?
I mean if the predicate is some 'not function' value the code will anyway throw an error when we try to use it as a function. (of course it is possible that the ASSERT gives us more information, but I doubt)